scripts/tools/tools-object.js

```diff
@@ -0,0 +1,57 @@
+const schema = require("./tools-schema.json");
+const axios = require('axios')
+var validate = require("jsonschema").validate;
+
+let appendData = {
+    generator: [],
+};
+
+const makeObject = (object, repositoryUrl, isAsyncAPIrepo) => {
+    let resultantObject = {
+      title: object.title,
+      description: object.description,
+      links: {
+        websiteUrl: object.links.websiteUrl,
+        docsUrl: object.links.docsUrl,
+        iconUrl: object.links.iconUrl,
+        repoUrl: repositoryUrl,
+      },
+      filters: {
+        languages: object.filters.language,
+        technology: object.filters.technology,
+        categories: object.filters.categories,
+        hasCommercial: object.filters.hasCommercial,
+        isAsyncAPIOwner: isAsyncAPIrepo,
+      },
+    };
+    return resultantObject;
+  };
+  
+  const getContent = async (result_object) => {
+    let reference_id = result_object.url.split("=")[1];
+    let download_url = `https://raw.githubusercontent.com/${result_object.repository.full_name}/${reference_id}/${result_object.path}`;
+    //   console.log(download_url);
+    let toolObject;
+    const resp = await axios.get(download_url);
```

Comment:
ok, but we can't define the type to `data` because it is not Typescript. This will show error in Javascript files. `{ data }` is sufficient I think?

Reply:
> ok, but we can't define the type to data because it is not Typescript. This will show error in Javascript files.

It won't because there you don't define TS types, but create alias. It's a way to create typings:

```ts
 const { data: toolFileContent }: { data: TYPE }
```

so you shouldn't worry about typings here.